c7n/mu.py

```diff
@@ -63,7 +65,7 @@ class PythonPackageArchive:
 
     zip_compression = zipfile.ZIP_DEFLATED
 
-    def __init__(self, modules=(), cache_file=None):
+    def __init__(self, modules=(), cache_file=None, layer_archive=False):
```

Comment:
is there any harm in generating the layer every time? otherwise this becomes a game of cat and mouse remembering to set the layer when botocore/boto3 is out of date

Reply:
I had thought of doing that originally but the generation of the layer can take some time roughly 15-20 seconds and since MU is instantiated for each policy it could add up to quite a bit of time eventually. I also wasn't sure what would happen at the layers level after a lot of versions had been built up. I'm not sure if they cap that storage or not.

Reply:
well, looking at the code it seems like it will try to find the layer with the name:

```
        layer_name = f'c7n-botocore{botocore.__version__} \
             -boto3{boto3.__version__}'.replace('.', '-')
```

in that case, you would only need to generate the layer once per deployment

Reply:
This is a best effort to ensure the layer is only generated once for each boto3/core combo, if the layer already exists it will skip it. The drawback there is an API call to check for the existence every time but I was having some trouble bubbling the layerARN up to be cached somewhere else.

Reply:
looks like the layer Arn is relatively easy to figure out, e.g. `arn:aws:lambda:<region>:<account id>:layer:<layer name>:<version>` example here: https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html you might be able to get away with assuming the name of the layer first and then if it doesn't exist to just create it